test/e2e/framework/pod/resource.go

```diff
@@ -544,28 +585,30 @@ func checkPodsCondition(c clientset.Interface, ns string, podNames []string, tim
 
 // GetPodLogs returns the logs of the specified container (namespace/pod/container).
 func GetPodLogs(c clientset.Interface, namespace, podName, containerName string) (string, error) {
-	return getPodLogsInternal(c, namespace, podName, containerName, false, nil)
+	return getPodLogsInternal(c, namespace, podName, containerName, false, nil, -1)
 }
 
 // GetPodLogsSince returns the logs of the specified container (namespace/pod/container) since a timestamp.
 func GetPodLogsSince(c clientset.Interface, namespace, podName, containerName string, since time.Time) (string, error) {
 	sinceTime := metav1.NewTime(since)
-	return getPodLogsInternal(c, namespace, podName, containerName, false, &sinceTime)
+	return getPodLogsInternal(c, namespace, podName, containerName, false, &sinceTime, -1)
 }
 
 // GetPreviousPodLogs returns the logs of the previous instance of the
 // specified container (namespace/pod/container).
 func GetPreviousPodLogs(c clientset.Interface, namespace, podName, containerName string) (string, error) {
-	return getPodLogsInternal(c, namespace, podName, containerName, true, nil)
+	return getPodLogsInternal(c, namespace, podName, containerName, true, nil, -1)
 }
 
 // utility function for gomega Eventually
-func getPodLogsInternal(c clientset.Interface, namespace, podName, containerName string, previous bool, sinceTime *metav1.Time) (string, error) {
+// if tailLen is -1, all the pod logs will be returned.
+func getPodLogsInternal(c clientset.Interface, namespace, podName, containerName string, previous bool, sinceTime *metav1.Time, tailLen int) (string, error) {
 	request := c.CoreV1().RESTClient().Get().
 		Resource("pods").
 		Namespace(namespace).
 		Name(podName).SubResource("log").
 		Param("container", containerName).
+		Param("tail", strconv.Itoa(tailLen)).
```

Comment:
ref: https://github.com/kubernetes/kubernetes/blob/ab609a418ea804c896cbcba4d93a77e46ca76dff/staging/src/k8s.io/api/core/v1/types.go#L5233-L5236

My read is you should be using `tailLines` and omitting it if you're passing in -1.